### PR TITLE
moving examples to docstrings

### PR DESCRIPTION
**Pull request by gferreira**
¡Warning! A few issues still need to be fixed:

1. with `::downloadcode`, special characters like `\n` and `\t` get converted into the actual newline and tab characters – this broke some examples in which those characters appear, like `indent.py` and `tabsFormattedString.py`

2. the ‘downloads’ folder is included in the git repo – this is redundant since the examples are now in the docstrings. (easy to fix by adding the ‘downloads’ folder to `.gitignore`)

3. the ‘downloads’ folder should probably be cleared when running `make html`, or the new download files should simply overwrite existing ones – otherwise you end up with `rect.py`, `rect1.py`, `rect2.py` etc.

### CODE REVIEW

**Comment by typemytype** [comment]
sphinx escapes "\n" with "\\n"

maybe not so nice for `help(fun)` but the download and docs works as it should

**Comment by gferreira** [comment]
are you sure the docs are ok? it looks like the sample code is breaking at `\n` here:

https://github.com/typemytype/drawbot/pull/89/commits/76e44c917c6371fabe66777469ef96e48947b4e0#diff-b2fc94f9cc832e46572e3e9d70505ed1

screen shot 2017-01-19 at 21 19 12

**Comment by typemytype** [comment]
double \\\n slashes

github was also escaping them :)

File: docs/downloads/indent.py [hunk 1]
```diff
@@ -38,29 +38,4 @@
 # add text
 t += txt
 # add hard return
-t += "\n"
```
Comment (typemytype) [inline docs/downloads/indent.py]: this should be `\\n` to get properly escaped

**Comment by gferreira** [comment]
thanks – fixed! (issues 2 & 3 remain though)

**Comment by gferreira** [comment]
https://github.com/typemytype/drawbot/pull/89/commits/20d2a663b7bd2ed9a0d9a03ba7041048202d7ec9 addresses point 2 (see above)

**Comment by typemytype** [comment]
super thanks!!!

**Pull request merged**